src/hooks/RouterHooks.js

```diff
@@ -29,47 +30,96 @@ export const useRedirectFromInstanceToSenarioView = () => {
   }, [currentWorkspaceData?.webApp?.options?.instanceView]);
 };
 
-export const useRedirectionToScenario = (sortedScenarioList) => {
-  const isMounted = useRef(false);
+export const useRouterScenarioId = () => {
   const routerParameters = useParams();
-  const currentScenario = useCurrentScenario();
+  return routerParameters?.scenarioId;
+};
+
+export const useRedirectionToScenario = () => {
+  const currentScenarioId = useCurrentScenarioId();
+
   const navigate = useNavigate();
   const handleScenarioChange = useFindScenarioById();
-  const updateCurrentScenario = useUpdateCurrentScenario();
+  const sortedScenarioList = useSortedScenarioList();
 
+  const routerScenarioId = useRouterScenarioId();
+
+  // no routerId and no current scenarioId > Navigate to the first scenario
   useEffect(() => {
-    if (sortedScenarioList.length !== 0) {
-      if (currentScenario?.data?.id === undefined) return;
-      if (routerParameters.scenarioId === undefined) {
-        navigate(`${currentScenario.data.id}`, { replace: true });
-      } else if (currentScenario.data.id !== routerParameters.scenarioId) {
-        handleScenarioChange(routerParameters.scenarioId);
-        if (!sortedScenarioList?.find((scenario) => scenario.id === routerParameters.scenarioId))
-          navigate(`${currentScenario.data.id}`);
-      }
+    if (sortedScenarioList.length !== 0 && !routerScenarioId && !currentScenarioId) {
+      navigate(sortedScenarioList[0].id);
     }
```

Comment:
useless brackets on if statement

Reply:
i keep it for readability (line too long)